decompiler/pipeline/controlflowanalysis/readability_based_refinement.py

```diff
@@ -219,6 +220,23 @@ def _requirement_without_reinitialization(ast: AbstractSyntaxTree, node: Abstrac
                 return True
 
 
+def removeGuardedDoWhile(ast: AbstractSyntaxTree):
+    """ Removes a if statement which guards a do-while loop when:
+            -> there is nothing in between the if-node and the do-while-node 
+            -> the if-node has only one branch
+            -> the condition of the branch is the same as the condition of the do-while-node
+    """
+    for loop_node in list(ast.get_loop_nodes_post_order()):
+        if not isinstance(loop_node, DoWhileLoopNode) or not isinstance(loop_node.parent.parent, ConditionNode):
+            continue
```

Comment:
Sorry, one more suggestion: Perhaps a function `get_potential_guarded_do_while_loops`. This would make this easier to understand. At one point, I tried to move all `isinstace` to the ast-classes, but I think it is okay here.

Reply:
Split the method in two + changed names to be more consistent